src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -274,6 +285,38 @@ boost::optional<GafferScene::EditScopeAlgo::TransformEdit> GafferScene::EditScop
 	};
 }
 
+const GraphComponent *GafferScene::EditScopeAlgo::transformEditReadOnlyReason( const Gaffer::EditScope *scope, const ScenePlug::ScenePath &path )
+{
+	auto *processor = const_cast<EditScope *>( scope )->acquireProcessor<SceneProcessor>( "TransformEdits", /* createIfNecessary = */ false );
+	if( !processor )
+	{
+		return MetadataAlgo::readOnlyReason( scope );
+	}
+
+	string pathString;
+	ScenePlug::pathToString( path, pathString );
+
+	auto *rows = processor->getChild<Spreadsheet::RowsPlug>( "edits" );
+	const Spreadsheet::RowPlug *row = rows->row( pathString );
+	if( !row )
+	{
+		return MetadataAlgo::readOnlyReason( rows );
+	}
+
+	// As any of our columns could be locked, pick the first one.
+	for( const auto child : { g_translate, g_rotate, g_scale, g_pivot } )
+	{
+		const Spreadsheet::CellPlug *cell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( child );
+		if( !cell ) { continue; }
+
+		if( const auto reason = MetadataAlgo::readOnlyReason( cell->valuePlug<V3fPlug>() ) )
+		{
+			return reason;
+		}
+	}
```

Comment:
I think we agreed that we'd drop the `!cell` defensiveness here, because nobody should be deleting them, and we should fix anything that attempts to. But I think this is also missing the fact that the individual plugs of a value could be read only. I think this version plugs that gap while being more concise and also robust to missing plugs without needing explicit checks :

```
for( const auto &p : Plug::RecursiveRange( *row->cellsPlug() ) )
{
    if( MetadataAlgo::readOnly( p.get() ) )
    {
        return p.get();
    }
}
```

Reply:
Much better, thanks John, sorry I'd temporarily forgotten about `RecursiveRange`. Improved [here](https://github.com/GafferHQ/gaffer/pull/3840/commits/de7a2934c5ede418ab0ac14ad9cafa7c9136fa98).